include/cantera/kinetics/RxnRates.h

```diff
@@ -290,8 +290,11 @@ class Plog
             }
             log_k2 = std::log(k);
         }
-
-        return std::exp(log_k1 + (log_k2-log_k1) * (logP_-logP1_) * rDeltaP_);
+        if (logP_ == logP1_) {
+            return std::exp(log_k1);
+        } else {
+            return std::exp(log_k1 + (log_k2-log_k1) * (logP_-logP1_) * rDeltaP_);
+        }
```

Comment:
Yes, I believe it is necessary(or at least there should be applied some change). I think in the validation case, since the input value of `update_C` is taken from `pressures_` in `Plog::validate`, so the lower reference pressure `log_P1`  should be the exact same value with `logP_` because the logarithm of lower reference pressure(`log_P1`) is assigned from `pressures_`  as well.  I add `if` here because I want to differentiate the validation case from the normal case(case that input pressure does not equal to any refference pressure). `std::exp(log_k1 + (log_k2-log_k1) * (logP_-logP1_) * rDeltaP_)` involves `log_k2` at higher reference pressure and `log_k1` at lower reference pressure, in validation case, every reference pressure point will be called twice as higher pressure point first and lower pressure point when the input changes to next reference pressure point. In this case, if the rate constant at a reference pressure point is invalid, the error will be raised twice as it involves in formula`std::exp(log_k1 + (log_k2-log_k1) * (logP_-logP1_) * rDeltaP_)` twice. For example, in reaction 
```
- equation: CH2CHOO <=> CO + CH3O  
  type: pressure-dependent-Arrhenius
  rate-constants:
  - {P: 1.0 atm, A: 1.8e+33, b: -7.27, Ea: 3.376e+04}
  - {P: 1.0 atm, A: 8.69e-50, b: 16.63, Ea: -3900.0}  
  - {P: 10.0 atm, A: -3.83e+33, b: -7.2, Ea: 3.51e+04}
  - {P: 10.0 atm, A: 1.19e-39, b: 13.61, Ea: -1317.0}
```
reaction rate constant at `P = 1.0 atm` is  valid (positive),  when validate the pressure point P = 1.0 atm, `log_k1` at lower reference pressure 1.0 atm is valid and it should pass the validation, but the `log_k2` at higher reference pressure 10.0 atm is invalid, so the formula `std::exp(log_k1 + (log_k2-log_k1) * (logP_-logP1_) * rDeltaP_)` will return `NaN` , which should actually be log_k1 because `logP_ - logP1_` equal to zero in the validation case.  In the original code, it is not a problem because is throws the next pressure point 10 atm in the error message and stops. However, to check all the reference pressure point in validation case, It is not necessary to involve `log_k2` when calculating `log_k1` .